src/r2mm/manager/SettingsDexieStore.ts

```diff
@@ -24,66 +22,70 @@ export default class SettingsDexieStore extends Dexie {
         });
 
         const store = {
-            value: `++id,settings`
+            value: `++id,settings`,
+            games: `identifier,settings`,
         } as any;
 
-        GameManager.gameList
-            .forEach(value => {
-                store[value.settingsIdentifier] = `++id,settings`;
-            });
+        // Setup the v3 (>= 75) dexie store.
+        // This contains the following tables:
+        // - `games`: (settingsIdentifier,settings) <-- Game-specific settings.
+        // - `value`: (++id, settings)              <-- Global settings.
+        this.version(75).stores(store).upgrade(async (trans) => {
+            // Migrate the current game's legacy (v2) table to the v3 schema, if applicable - versions < 75.
+            //
+            // v2 (whose types we still utilize) stored game-specific settings in individual tables.
+            // This worked but required us to bump the Dexie store version every time we added a new game. Yuck!
+            // To get around this we have moved all game settings into the `games` table, where the key
+            // is the settings identifier of the game, and the value is a JSON string.
+            const ignore = ["value", "games"];
+            const legacyTables = trans.storeNames.filter(x => !ignore.includes(x)).map(x => trans.table(x));
```

Comment:
IDK if I'm missing something here, but on my local env `trans.storeNames` contains only "value" and "games", so none of the existing game settings are migrated. Based on a very quick check I couldn't find any way to access the old tables via `trans`, although logging the object shows that it's at least on some level aware of the old tables. So should we loop over the `GameManager.gameList` here too?

Reply:
Applied your suggestion, worked.